packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -824,46 +836,52 @@ function ResizableTableColumnHeader(props) {
               <div className={classNames(styles, 'spectrum-Table-headerCellText')}>{column.rendered}</div>
             }
             {
-              columnProps.allowsResizing && <ChevronDownMedium UNSAFE_className={classNames(styles, 'spectrum-Table-menuChevron')} />
+              (columnProps.allowsResizing || hasCustomActions) && <ChevronDownMedium UNSAFE_className={classNames(styles, 'spectrum-Table-menuChevron')} />
             }
           </TableColumnHeaderButton>
-          <Menu onAction={onMenuSelect} minWidth="size-2000" items={items}>
-            {(item) => (
-              <Item>
-                {item.label}
-              </Item>
-            )}
-          </Menu>
-        </MenuTrigger>
-        <Resizer
-          ref={resizingRef}
-          column={column}
-          showResizer={showResizer}
-          onResizeStart={onResizeStart}
-          onResize={onResize}
-          onResizeEnd={onResizeEnd}
-          triggerRef={useUnwrapDOMRef(triggerRef)} />
-        <div
-          aria-hidden
-          className={classNames(
-            styles,
-            'spectrum-Table-colResizeIndicator',
+          <Menu {...menuProps} onAction={onMenuSelect} minWidth="size-2000">
```

Comment:
One issue is we are assuming the `menu` element can only be a `<Menu>` and not a wrapper or something which would get thrown away here. Not sure if this will be a problem?

Reply:
yeah, noted here as well https://github.com/adobe/react-spectrum/pull/3886#issuecomment-1379657688 along with some thoughts I had on how to address it